spring-kafka-test/src/main/java/org/springframework/kafka/test/EmbeddedKafkaKraftBroker.java

```diff
@@ -243,6 +247,37 @@ private void start() {
 		System.setProperty(SPRING_EMBEDDED_KAFKA_BROKERS, getBrokersAsString());
 	}
 
+	private static void setConfigProperty(Object clusterBuilder, Object key, Object value) {
+		try {
+			boolean isKafka39OrLater = isClassicConsumerPresent();
+			Class<?> builderClass = clusterBuilder.getClass();
+
+			if (isKafka39OrLater) {
+				// For Kafka 3.9.0+: setConfigProp(String, Object)
+				Method setConfigMethod = builderClass.getMethod("setConfigProp", String.class, Object.class);
+				setConfigMethod.invoke(clusterBuilder, (String) key, value);
```

Comment:
Also see `ReflectionUtils.findMethod()` and `ReflectionUtils.invokeMethod()`.
You won't need to deal with any exceptions then!